File: solve/0401_크레인 인형뽑기 게임/postforty/크레인인형뽑기게임.js
function solution(board, moves) {
  var answer = 0;

  const tempArr = [];

  for (let i of moves) {
    const idx = i - 1;
    for (let j of board) {
      if (j[idx] !== 0) {
        if (tempArr[tempArr.length - 1] === j[idx]) {
          tempArr.pop();
          answer += 2;
        } else {
          tempArr.push(j[idx]);
        }
        j[idx] = 0;
        break;
      }
    }
  }

  return answer;
}

console.log(
  solution(
    [
      [0, 0, 0, 0, 0],
      [0, 0, 1, 0, 3],
      [0, 2, 5, 0, 1],
      [4, 2, 4, 4, 2],
      [3, 5, 1, 3, 1],
    ],
    [1, 5, 3, 5, 1, 2, 1, 4]
  )
);

File: solve/0401_크레인 인형뽑기 게임/postforty/크레인인형뽑기게임.py
def solution(board, moves):
    answer = 0

    temp_list = []

    for i in moves:
        idx = i - 1
        for j in board:
            if j[idx] != 0:
                if not temp_list:
                    temp_list.append(j[idx])
                    j[idx] = 0
                    break
                else:
                    if temp_list[-1] == j[idx]:
                        temp_list.pop()
                        answer += 2
                    else:
                        temp_list.append(j[idx])
                j[idx] = 0
                break
                # try:
                #     temp_list[-1]
                # except:
                #     temp_list.append(j[idx])
                #     j[idx] = 0
                #     break
                # else:
                #     if temp_list[-1] == j[idx]:
                #         temp_list.pop()
                #         answer += 2
                #     else:
                #         temp_list.append(j[idx])
                #     j[idx] = 0
                #     break
                    
    return answer

print(solution([
      [0, 0, 0, 0, 0],
      [0, 0, 1, 0, 3],
      [0, 2, 5, 0, 1],
      [4, 2, 4, 4, 2],
      [3, 5, 1, 3, 1],
    ],
    [1, 5, 3, 5, 1, 2, 1, 4]))